inginious/frontend/user_manager.py

```diff
@@ -288,15 +290,25 @@ def auth_user(self, username, password):
         Authenticate the user in database
         :param username: Username/Login
         :param password: User password
-        :return: Returns a dict represrnting the user
+        :return: Returns a dict representing the user
         """
-        password_hash = hashlib.sha512(password.encode("utf-8")).hexdigest()
+        password_hash = hash_content(password)
 
         user = self._database.users.find_one(
             {"username": username, "password": password_hash, "activate": {"$exists": False}})
 
         return user if user is not None and self.connect_user(username, user["realname"], user["email"],
-                                                              user["language"], user.get("tos_accepted", False)) else None
+                                                              user["language"],
+                                                              user.get("tos_accepted", False)) else None
+
+    def is_user_activated(self, username):
+        """
+        Verify if user is activated
+        :param username: Username/Login
```

Comment:
You should clearly allow for several usernames as you currently perform one request per user in a loop.